Sources/ApolloAPI/GraphQLOperation.swift

```diff
@@ -6,33 +6,92 @@ public enum GraphQLOperationType {
+  /// [APQs](https://www.apollographql.com/docs/apollo-server/performance/apq).
+  ///
+  /// This allow the operation definition to be persisted using an `operationIdentifier` instead of
+  /// being sent with every operation request. If the server does not recognize the
+  /// `operationIdentifier`, the network transport can send the provided definition to
+  /// "automatically persist" the operation definition.
+  case automaticallyPersisted(operationIdentifier: String, definition: OperationDefinition)
+
+  /// Provides only the `operationIdentifier` for operations that have been previously persisted
+  /// to an Apollo Server using
+  /// [APQs](https://www.apollographql.com/docs/apollo-server/performance/apq).
+  ///
+  /// If the server does not recognize the `operationIdentifier`, the operation will fail. This
+  /// method should only be used if you are manually persisting your queries to an Apollo Server.  
+  case persistedOperationsOnly(operationIdentifier: String)
+}
+
+/// The definition of an operation to be provided over network transport.
+///
+/// This data represents the `Definition` for a `Document` as defined in the GraphQL Spec.
+/// In the case of the Apollo client, the definition will always be an `ExecutableDefinition`.
+/// - See: [GraphQLSpec - Document](https://spec.graphql.org/draft/#Document)
+public struct OperationDefinition {
+  let operationDefinition: String
+  let fragments: [Fragment]?
+
+  public init(_ definition: String, fragments: [Fragment]? = nil) {
+    self.operationDefinition = definition
+    self.fragments = fragments
+  }
+
+  public var queryDocument: String {
+    var document = operationDefinition
+    fragments?.forEach {
+      document.append("\n" + $0.fragmentDefinition)
+    }
+    return document
+  }
+}
+
 public protocol GraphQLOperation: AnyObject {
   typealias Variables = [String: GraphQLOperationVariableValue]
 
-  var operationType: GraphQLOperationType { get }
-
-  var operationDefinition: String { get }
-  var operationIdentifier: String? { get }
   var operationName: String { get }
-
-  var queryDocument: String { get }
+  var operationType: GraphQLOperationType { get }
+  var document: DocumentType { get }
 
   var variables: Variables? { get }
 
   associatedtype Data: RootSelectionSet
 }
 
 public extension GraphQLOperation {
-  var queryDocument: String {
-    return operationDefinition
+  var variables: Variables? {
+    return nil
   }
 
-  var operationIdentifier: String? {
-    return nil
+  var definition: OperationDefinition? {
+    switch self.document {
+    case let .automaticallyPersisted(_, definition), let .notPersisted(definition):
+      return definition
+    default: return nil
+    }
   }
 
-  var variables: Variables? {
-    return nil
+  var operationIdentifier: String? {
+    switch self.document {
+    case let .automaticallyPersisted(identifier, _), let .persistedOperationsOnly(identifier):
```

Comment:
I generally prefer one case per line here since it's easier to parse while reading and also makes changes more obvious in a diff. Not a hill I'm gonna die on though

Reply:
Fixing this now.